src/molecule.cpp

```diff
@@ -142,24 +146,44 @@ Molecule::Molecule(LAMMPS *lmp, int narg, char **arg, int &index) :
 
   Molecule::initialize();
 
+  // check whether reading a json file or normal file
+
+  char *filename = arg[fileiarg];
+  int len = strlen(filename);
+  int jflag = 0;
+  char *file_ext = strstr(filename, "json");
+  
+  // check the file extension 
+
+  if(file_ext != NULL && strcmp(file_ext, "json") == 0)
+    jflag = 1;
+
   // scan file for sizes of all fields and allocate storage for them
 
-  if (me == 0) {
-    fp = fopen(arg[fileiarg], "r");
-    if (fp == nullptr)
-      error->one(FLERR, fileiarg, "Cannot open molecule file {}: {}", arg[fileiarg],
-                 utils::getsyserror());
+  if (jflag) { 
+    ifstream f(filename);
```

Comment:
It is a bad idea to have file I/O on *all* MPI ranks. LAMMPS may be running across 10s of thousands of processors. Having them all read the same file puts a huge burden on the networked file system of such machines. Instead file I/O should be done only on MPI rank 0 (as done with the conventional molecule file reader) and then either the data should be read and broadcase or read and parsed and the parsed results broadcast.